crypto/ex_data.c

```diff
@@ -362,6 +362,36 @@ void CRYPTO_free_ex_data(int class_index, void *obj, CRYPTO_EX_DATA *ad)
     ad->sk = NULL;
 }
 
+/*
+ * Allocate a given CRYPTO_EX_DATA item using the class specific allocation
+ * function
+ */
+int CRYPTO_alloc_ex_data(int class_index, void *obj, CRYPTO_EX_DATA *ad,
```

Comment:
If this function is going into the public API it needs documentation. (Should it be in the public API?)

Reply:
The rest of the ex_data stuff is public, so.

Thanks for the reminder, I'll document

Reply:
Eh gods, I had forgotten how bad the documentation for the ex_data stuff is.  It needs a complete rewrite, which I'd rather make subject for another PR.  I'm still mentioning this function, but am doing so as much in passing as `CRYPTO_set_ex_data`.

Reply:
There needs to be a test for this new function too